README.md

```diff
@@ -166,42 +167,75 @@ Alternatively you are able to use any `F[_]` which is convertible to/from `monix
 The Scala API uses types-conversions for both consumer and producer, that means you don't have to work directly with `Bytes` (however you
 still can, if you want) and you touch only your business class which is then (de)serialized using provided converter.
 
+Since the API is fully effectful don't forget
+
 The library uses two types of executors - one is for blocking (IO) operations and the second for callbacks. You _have to_ provide both of them:
 1. Blocking executor as `ExecutorService`
 1. Callback executor as `scala.concurrent.ExecutionContext`
 
 ```scala
-import com.typesafe.config.ConfigFactory
-import com.avast.metrics.api.Monitor
-import com.avast.clients.rabbitmq._ // for generic types support
+import java.util.concurrent.{ExecutorService, Executors}
+
 import com.avast.bytes.Bytes
+import com.avast.clients.rabbitmq._
+import com.avast.clients.rabbitmq.api.{Delivery, DeliveryResult, RabbitMQConsumer, RabbitMQProducer}
+import com.avast.metrics.scalaapi.Monitor
+import com.typesafe.config.ConfigFactory
 import monix.eval._
+import monix.execution.Scheduler
+
+import scala.concurrent.duration.Duration
 
 val config = ConfigFactory.load().getConfig("myRabbitConfig")
 
-implicit val ec: ExecutionContext = ???
+implicit val sch: Scheduler = ???
```

Comment:
On one hand you are using `???`, on the other hand you actually create a cached thread pool? Either use `???` everywhere or better provide some sane defaults so that the code can just be copied and used.